src/systems/filecoin_mining/storage_mining/storage_miner_actor_state.go

```diff
@@ -1,16 +1,54 @@
 package storage_mining
 
 import (
+	"errors"
+
+	ipld "github.com/filecoin-project/specs/libraries/ipld"
+	spc "github.com/filecoin-project/specs/systems/filecoin_blockchain/storage_power_consensus"
 	block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
 	deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
 	sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+	addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+	stateTree "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
+	util "github.com/filecoin-project/specs/util"
 )
 
 func (st *StorageMinerActorState_I) _isChallenged() bool {
 	return st.ChallengeStatus().IsChallenged()
 }
 
-func (st *StorageMinerActorState_I) _canBeElected(epoch block.ChainEpoch) bool {
+func (st *StorageMinerActorState_I) _getStoragePowerActorState(stateTree stateTree.StateTree) spc.StoragePowerActorState {
+	powerAddr := addr.StoragePowerActorAddr
+	actorState := stateTree.GetActorState(powerAddr)
+	substateCID := actorState.State()
+
+	substate, subErr := node.LocalGraph().Get(ipld.CID(substateCID))
+	if subErr != nil {
+		panic("No CID error")
+	}
+
+	// TODO fix conversion to bytes
+	panic(substate)
+	var serializedSubstate util.Serialization
+	spa, err := spc.Deserialize_StoragePowerActorState(serializedSubstate)
+
+	if err == nil {
+		panic("Deserialization error")
+	}
+	return spa
+}
```

Comment:
Blocking: I believe this pattern of inspecting another actor's state should be explicitly banned. Write a method/message to query the SPA for the desired information. In this case I think this entire addition should migrate to SPA.